### DISCUSSION TITLE
求助：Verify Alipay Response Sign Failed

### DISCUSSION THREAD

**Discussion by yanghaitao3864** [yansongda/pay#951] [Q&A]
请求支付时验签方法中的sign为空值，返回了这个错误：Verify Alipay Response Sign Failed
protected function verifySign(Rocket $rocket): void
    {
        $response = $rocket->getDestination();
        $result = $response->get($this->getResultKey($rocket->getPayload()));
        $sign = $response->get('sign', '');
        write_log('$sign',$sign);
        write_log('$sign',$result);
        if ('' === $sign || is_null($result)) {
            throw new InvalidResponseException(Exception::INVALID_RESPONSE_SIGN, 'Verify Alipay Response Sign Failed', $response);
        }

        verify_alipay_sign($rocket->getParams(), json_encode($result, JSON_UNESCAPED_UNICODE), base64_decode($sign));
    }

**Comment by yansongda** [comment]
可以看下 `$response` 是啥？
